Authentication/AuthenticationBootstrapper.swift

```diff
@@ -12,8 +12,9 @@ import Foundation
     Bootstrapper to start the application.
     Takes care of starting the authentication process before the main View Controller of the app when necessary,
     and after the user logs out.
+    The authentication process includes login, signup and recover password logic.
 */
-public class AuthenticationBootstraper<User: UserType, SessionService: SessionServiceType where SessionService.User == User> {
+public class AuthenticationBootstrapper<User: UserType, SessionService: SessionServiceType where SessionService.User == User> {
```

Comment:
So then we can do it with the controller delegate too. We leave the delegates as properties to be able to customize some changes without inheriting, but if they are so many changes or important ones, the user will need to inherit and override. Sounds good?